src/server/test/routes/logsRouteTests.js

```diff
@@ -0,0 +1,302 @@
+           const logTypes = testLogs.map(log => log.logType).sort();
+           expect(logTypes).to.deep.equal(['INFO', 'WARN']);
+       });
+
+       mocha.it('should respect log limit parameter', async () => {
+           const messagePrefix = 'Limit test message';
+
+           // Insert multiple logs
+           for (let i = 0; i < 5; i++) {
+               await chai.request(app)
+                   .post('/api/logs/info')
+                   .set('token', token)
+                   .send({ message: `${messagePrefix} ${i}` });
+           }
+
+           // Query with limit of 2
+           const response = await chai.request(app)
+               .get('/api/logs/logsmsg/getLogsByDateRangeAndType')
+               .set('token', token)
+               .query({ 
+                   timeInterval: moment().subtract(1, 'minute').toISOString(),
+                   logTypes: 'INFO',
+                   logLimit: '2'
+               });
+
+           expect(response.status).to.equal(200);
+           expect(response.body).to.be.an('array');
+           
+           // Should not return more than the limit
+           expect(response.body.length).to.be.at.most(2);
+       });
+
+       mocha.it('should handle date range filtering', async () => {
+           const oldMessage = 'Old test message';
+           const recentMessage = 'Recent test message';
+
+           // Insert a log
+           await chai.request(app)
+               .post('/api/logs/info')
+               .set('token', token)
+               .send({ message: oldMessage });
+
+           // Wait a moment, then insert another
+           await new Promise(resolve => setTimeout(resolve, 1000));
+           const cutoffTime = moment();
+           await new Promise(resolve => setTimeout(resolve, 1000));
+
+           await chai.request(app)
+               .post('/api/logs/info')
+               .set('token', token)
+               .send({ message: recentMessage });
+
+           // Query for logs after the cutoff time
+           const response = await chai.request(app)
+               .get('/api/logs/logsmsg/getLogsByDateRangeAndType')
+               .set('token', token)
+               .query({ 
+                   timeInterval: cutoffTime.toISOString(),
+                   logTypes: 'INFO',
+                   logLimit: '100'
+               });
+
+           expect(response.status).to.equal(200);
+           const testLogs = response.body.filter(log => 
+               log.logMessage === oldMessage || log.logMessage === recentMessage
+           );
+
+           // Should only contain the recent message
+           expect(testLogs).to.have.lengthOf(1);
+           expect(testLogs[0].logMessage).to.equal(recentMessage);
+       });
+
+       mocha.it('should return empty array when no logs match filters', async () => {
+           // Query for logs with a very specific type that doesn't exist
+           const response = await chai.request(app)
```

Comment:
I'm wondering if this would be a stronger test if other log msgs were inserted so some existed that are not returned.